indigo/analysis/refs/base.py

```diff
@@ -123,3 +123,132 @@ class RefsFinderENG(BaseRefsFinder):
+        if candidate_elements:
+            element_id = candidate_elements[0].get('id')
+            return f'#{element_id}'
+
+    def is_internal(self, match):
+        ref = match.group("ref")
+        if ref.endswith('the ') or ref.endswith('Act '):
+            return False
+        return True
+
+    def find_references(self, root):
+        for ancestor in self.ancestor_nodes(root):
+            for candidate in self.candidate_nodes(ancestor):
+                node = candidate.getparent()
+
+                if not candidate.is_tail:
+                    # text directly inside a node
+                    match = self.section_re.search(node.text)
+                    if match and self.is_internal(match):
+                        # mark the reference and continue to check the new tail
+                        node = self.mark_reference(node, match, in_tail=False)
+
+                while node is not None and node.tail:
+                    match = self.section_re.search(node.tail)
+                    if (not match) or (not self.is_internal(match)):
+                        break
+
+                    # mark the reference and continue to check the new tail
+                    node = self.mark_reference(node, match, in_tail=True)
+
+    def mark_reference(self, node, match, in_tail):
+        # first check to see if a reference will be found
+        href = self.make_href(node, match)
+        if not href:
+            return node
+
+        ref, start_pos, end_pos = self.make_ref(node, match)
+
+        if in_tail:
+            node.addnext(ref)
+            node.tail = match.string[:start_pos]
+            ref.tail = match.string[end_pos:]
+        else:
+            node.text = match.string[:start_pos]
+            node.insert(0, ref)
+            ref.tail = match.string[end_pos:]
+
+        return ref
+
+    def make_ref(self, node, match):
+        """ Make a reference out of this match, returning a (ref, start, end) tuple
+        which is the new ref node, and the start and end position of what text
+        in the parent element it should be replacing.
+        """
+        ref = etree.Element(self.ref_tag)
+        ref.text = match.group('num')
+        ref.set('href', self.make_href(node, match))
+        return ref, match.start('num'), match.end('num')
+
+    def ancestor_nodes(self, root):
+        for x in self.ancestor_xpath(root):
+            yield x
+
+    def candidate_nodes(self, root):
+        for x in self.candidate_xpath(root):
+            yield x
+
+
+@plugins.register('section-refs')
+class SectionRefsFinderENG(BaseSectionRefsFinder):
+    """ Finds internal references to sections in documents, of the form:
+
+        section 26
+        TODO: add more patterns
+
+    """
+
+    # country, language, locality
+    locale = (None, 'eng', None)
+
+    section_re = re.compile(
+        r'''\b[sS]ections?\s+
+            (?P<ref>
+             (?P<num>\d+)
```

Comment:
What about "Section 26b"?

Perhaps also a test case ensuring that "subsection 26(b)" is not matched? Your regex looks fine, but a test is a good idea.